packages/secure_storage/amplify_secure_storage_dart/lib/src/types/secure_storage_factory.dart

```diff
@@ -0,0 +1,21 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
+
+/// A factory constructor for a [SecureStorageInterface] instance.
+typedef SecureStorageFactory = SecureStorageInterface Function(
+  AmplifyStorageScope amplifyScope,
+);
+
+/// Storage scope for config of [SecureStorageInterface] within Amplify.
+enum AmplifyStorageScope {
+  /// Storage scope for Auth plugin using Cognito.
+  awsCognitoAuthPlugin,
```

Comment:
Not in love with the naming of these. How about `auth`, `analytics`, and `pushNotifications`?

Reply:
I believe I had used `awsCognitoAuthPlugin` because that was what native was using for credential storage. We don't have to align with that, but including something related to Cognito in the name seems like a good idea to avoid collisions (even though that is not really possibly right now with only one auth plugin).

If/when we have a second auth plugin, I think we would want them to use different scopes, right? A customer probably wouldn't use two auth plugins at once, but they may migrate from one to the other.

Reply:
Ah true, true.. okay, yeah maybe I'd suggest something a bit shorter then like `authCognito`. The scope string can be the full name, but maybe just for the enum so it appears cleaner where it shows up.

Reply:
Thanks for the insights.  

We only use the enum in one place (auth_plugin_impl) so is it really that necessary to shorten? 

I prefer having the enum names mapping directly to their value when possible to reduce logic.